### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by agaffney**

### CODE REVIEW

File: .github/workflows/ci-docker.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Docker CI
+
+on:
+  pull_request:
+    branches: ['main']
+    paths: ['Dockerfile','cmd/**','docs/**','internal/**','go.*','.github/workflows/ci-docker.yml']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: blinklabs/snek
```
Comment (wolf31o2) [inline .github/workflows/ci-docker.yml:10]: This is fine for here, but publishing will need to be `blinklabs-io/snek` when we actually push.

**Review by wolf31o2 — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+build: $(BINARIES)
+
+mod-tidy:
+	# Needed to fetch new dependencies and add them to go.mod
+	go mod tidy
+
+clean:
+	rm -f $(BINARIES)
+
+test:
```
[the remark this answers is not in the document]
Comment (agaffney) [reply Makefile]: Done.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.18 AS build
+
+WORKDIR /code
+COPY . .
+RUN make build
+
+FROM ubuntu:focal AS final
```
[the remark this answers is not in the document]
Comment (agaffney) [reply Dockerfile]: Done

File: .github/workflows/ci-docker.yml (hunk 1 above)
Comment (agaffney) [reply .github/workflows/ci-docker.yml]: I fixed this anyway

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**